sdk/core/azure-core/inc/azure/core/datetime.hpp

```diff
@@ -50,60 +189,52 @@ namespace Azure { namespace Core {
       /// RFC 1123.
       Rfc1123,
 
-      /// ISO 8601.
-      Iso8601,
+      /// RFC 3339.
+      Rfc3339,
```

Comment:
Based on this (https://github.com/Azure/azure-sdk-for-cpp/issues/776#issuecomment-724582360), it looks like we need to support the extended ISO 8601-1:2019 profile with all levels of granularity, and not just the RFC 3339 subset.

That is what the current implementation is doing anyway as shown by this test:
https://github.com/Azure/azure-sdk-for-cpp/blob/a2ddbc8cc913ff123c186bb32ab8b72f8d6265f0/sdk/core/azure-core/test/ut/datetime.cpp#L47-L53

For example, according to RFC 3339, this is invalid: `"yyyy'-'MM'-'dd"`

Therefore, we should revert this back to ISO.

```suggestion
      /// ISO 8601.
      Iso8601,
```

For additional context:
https://docs.microsoft.com/en-us/dotnet/standard/datetime/system-text-json-support#support-for-parsing

Reply:
BTW, our implementation would allow it, see line 47 in the unit tests:
```
    auto dt = DateTime::Parse("2013-05-17", DateTime::DateFormat::Rfc3339);
    EXPECT_NE(0, static_cast<DateTime::Duration>(dt).count());
```

Reply:
> What makes you think that 'yyyy'-'MM'-'dd' is invalid for RFC 3339?

Because of what the ABNF says and things like:
https://validator.w3.org/feed/docs/error/InvalidRFC3339Date.html
> Note that RFC 3339 is more strict than the W3C Date and Time Format. In particular, all portions of the date and time must be present, with the exception of the fractional portions of the second, which may be omitted.

https://stackoverflow.com/a/522281/12509023
> Most notably RFC 3339 specifies a complete representation of date and time (only fractional seconds are optional).

But maybe that's semantics, and maybe our parser can be more lenient and allow it. That would be OK.

Its possible the strict adherence to what RFC 3339 is what a customer expects, as a feature, of our DateTime, so either we reject 2020-11-12, or allow those sub-forms, and document it explicitly. In that case, it seems strange to use that name for the enum value. The Cpp Rest SDK called it ISO 8601 and supported that flexibility in parsing. Why is that?

Reply:
OK, well the Azure HTTP guidelines explicitly state to use RFC 3339 and so I think we can expect the time portion and fail if the string is not truly RFC 3339.

Reply:
I don't see any value on failing when just "yyyy-MM-dd" is provided, and I also don't see anything in the links above that could mean that yyyy-MM-dd is incorrect.

Reply:
> I also don't see anything in the links above that could mean that yyyy-MM-dd is incorrect.

Hmm... 

@antkmsft - I quoted the relevant snippets in my post above:

>  In particular, **all portions** of the date and time **must be present**